src/app/clusters/group-key-mgmt-server/group-key-mgmt-server.cpp

```diff
@@ -443,8 +473,9 @@ bool emberAfGroupKeyManagementClusterKeySetWriteCallback(
         // supported by the server, because it is ... a new value unrecognized
         // by a legacy server, then the server SHALL generate a general
         // constraint error
-        commandObj->AddStatusAndLogIfFailure(commandPath, Status::ConstraintError,
+        chip::app::CommandHandler::LogStatus(commandPath, Status::ConstraintError,
```

Comment:
The whole point of AddStatusAndLogIfFailure, as far as I can tell, was to make it really easy for people to log the thing they are responding with.  The new setup requires repeating the same status twice, can lead to them diverging, etc.  Please leave AddStatusAndLogIfFailure bits as they were.

Reply:
I made the change intentional: a method with `And` in the name is a code smell and it should not have existed in the first place. This method encouraged awkward code:
  - calling this method with a failing status just for the log sideffect, when there is little benefit over clear addstatus and logging
  - logging with Status code using this method instead of the underlying CHIP_ERROR (and thus losing the extra info that is contained in CHIP_ERROR).
  
  I added an equivalent `Log` to be explicit and do one thing only.

Reply:
I know what you did but given that you are taking @tcarmelveilleux's PR and explicitly removing a thing he added in a previous PR, the whole thing is very odd.

What I would suggest is that we fix the actual bug in this PR, and move any refactoring of the AddStatus bits to separate PRs that can be reviewed separately (and can wait for @tcarmelveilleux to get back from vacation, whereas this one we need to fix so we stop failing tests).

Reply:
Ok, reverted that change, will do any updates in a separate PR.